### ISSUE TITLE
netcdf: segmentation fault when creating a file in a non-existing directory

### ISSUE THREAD

**Issue by rouault** [msys2/MINGW-packages#11918]
This is a regression due to the update to 4.9.0 done in https://github.com/msys2/MINGW-packages/pull/11879. Previous 4.8.1-4 package errors out cleanly. Issue found on the GDAL autotest continuous integration (cf https://github.com/OSGeo/gdal/issues/5970)

```
$ nccopy --version
[...]
netCDF library version 4.9.0 of Jun 21 2022 18:22:22 $
$ nccopy byte_no_cf.nc "\non_existing_path\out.nc"
Error:NCpathcvt: stat=-36 (NetCDF: Invalid argument)
Segmentation fault
```
where byte_no_cf.nc is in the attached zip (but true for any netCDF file). 
[byte_no_cf.nc.zip](https://github.com/msys2/MINGW-packages/files/8967630/byte_no_cf.nc.zip)

vs
```
$ nccopy --version
[...]
netCDF library version 4.8.1 of Apr  2 2022 18:28:59 $
$ nccopy byte_no_cf.nc "\non_existing_path\out.nc"
No such file or directory
Location: file ; line 2016
```

**Comment by mmuetzel** [comment]
Is hopefully fixed with PR #11920.
@rouault: Could you please check with the build artifacts from that PR (when they are ready) whether that resolves the error for you?
Should be available here (if building is successful):
https://github.com/msys2/MINGW-packages/actions/runs/2550818176

**Comment by mmuetzel** [comment]
Tried to upstream the change. But they seem to have elaborate requirements for contributions. If you already went through that process, feel free to submit a PR to them.

**Comment by rouault** [comment]
> Tried to upstream the change. But they seem to have elaborate requirements for contributions. If you already went through that process, feel free to submit a PR to them.

I've already submited PRs, but don't recall the CLA thing was in place (or maybe I just don't remember). Looking quickly, it seems it is just a matter of clicking on a few buttons to accept the CLA. I could do it if that's not practical for you

**Comment by mmuetzel** [comment]
Yes, please do. Those pages don't want to load correctly for. (And I don't understand why they want to collect my address, phone number, ...)

**Comment by mmuetzel** [comment]
I thought I already gave the permission with the comments here. But to be completely explicit with no room of doubt:
I don't claim any ownership on that one-liner modifying the code of netcdf.
